FormidableLabs/envy
# Replace Mock data button with debug menu for more developer options

## What does this PR do

- Adds a new `Menu` component which allows actions to be grouped in a dropdown style menu
- Replaces the development mode "Mock data" button with a "Debug menu" which includes:
  - Ability to inject mock traces (as per the previous "Mock data" button)
  - Ability to print current traces to the browser console for better raw data inspection
- Adds unit tests to retain 100% coverage

**Note: this is intended as a debug menu for the development of Envy, not as a feature of Envy to be used when monitoring and debugging traces of another application.  If this isn't clear, we can change how this is presented to be clearer.  This "Debug menu" will not appear in production builds of Envy**

## Screenshots

![image](https://github.com/FormidableLabs/envy/assets/17857833/4c9973f5-443b-4f72-876f-c57e14d4ce36)

## How to test

- Run `yarn workspace @envyjs/webui start`
- Verify that the "Debug menu" is displayed and performs the above actions
